apiserver/failoverservice/failoverimpl.go

```diff
@@ -179,10 +179,20 @@ func QueryFailover(name, ns string) msgs.QueryFailoverResponse {
 	return response
 }
 
-func validateClusterName(clusterName, ns string) (*crv1.Pgcluster, error) {
+//validateCluster validates the cluster name, as well as checks to see
+//if the referenced cluster needs to be upgraded
+func validateCluster(clusterName, ns string) (*crv1.Pgcluster, error) {
```

Comment:
Yeah, this is a bit confused. The function, before this modification, was:
```
func validateClusterName(clusterName, ns string) (*crv1.Pgcluster, error) {
	cluster := crv1.Pgcluster{}
	found, err := kubeapi.Getpgcluster(apiserver.RESTClient,
		&cluster, clusterName, ns)
	if !found {
		return &cluster, errors.New("no cluster found named " + clusterName)
	}

	return &cluster, err
}
```
And, really, is just getting the pgcluster. In most of our other implementations, we'd just put the call to `Getpgcluster` in the main function. The upgrade check here would normally also be in the main function. I'm happy to restructure this however we'd like to, or at very least revert the name if that is preferrable, but the upgrade check will need to be placed somewhere in this file.

Reply:
Either or, it should be modified :wink:

Reply:
After thinking this over, since this is used both during the creation of a failover and when querying for failover targets, it probably makes more sense to leave the original function as is, and add the 'is upgraded' checking for when an actual failover is trying to be initiated. 

This would bring this more in line with what I've done in the other implementations, specifically, allow 'informational' commands that make no modifications to execute against non-upgraded clusters. I'll test this out to make sure there are no surprises though....